src/status_im2/contexts/syncing/scan_sync_code/view.cljs

```diff
@@ -0,0 +1,230 @@
+      (rf/dispatch [:toasts/upsert
+                    {:icon           :i/info
+                     :icon-color     colors/danger-50
+                     :override-theme :light
+                     :text           (i18n/label :t/error-this-is-not-a-sync-qr-code)}]))))
+
+(defn render-background
+  [background show-camera? qr-view-finder]
+  (if (and show-camera? (:x qr-view-finder))
+    [hole-view/hole-view
+     {:style style/absolute-fill
+      :holes [(merge qr-view-finder
+                     {:borderRadius 16})]}
+     background]
+    background)
+)
+
+(defn view
+  [{:keys [title show-bottom-view? background]}]
+  (let [active-tab                (reagent/atom 1)
+        qr-view-finder            (reagent/atom {})
+        request-camera-permission (fn []
+                                    (rf/dispatch
+                                     [:request-permissions
+                                      {:permissions [:camera]
+                                       :on-allowed  #(reset! camera-permission-granted? true)
+                                       :on-denied   #(rf/dispatch
+                                                      [:toasts/upsert
+                                                       {:icon :i/info
+                                                        :icon-color colors/danger-50
+                                                        :override-theme :light
+                                                        :text (i18n/label
+                                                               :t/camera-permission-denied)}])}]))]
+    [:f>
+     (fn []
+       (let [insets            (safe-area/use-safe-area)
+             camera-ref        (atom nil)
+             read-qr-once?     (atom false)
+             scan-qr-code-tab? (= @active-tab 1)
+             show-camera?      (and scan-qr-code-tab? @camera-permission-granted?)
+             on-read-code      (fn [data]
+                                 (when-not @read-qr-once?
+                                   (reset! read-qr-once? true)
+                                   (js/setTimeout (fn []
+                                                    (reset! read-qr-once? false))
+                                                  3000)
+                                   (check-qr-code-data data)))]
+         (rn/use-effect
+          (fn []
+            (when-not @camera-permission-granted?
+              (permissions/permission-granted? :camera
+                                               #(reset! camera-permission-granted? %)
+                                               #(reset! camera-permission-granted? false)))))
+         [:<>
+          [render-background background show-camera? @qr-view-finder]
+          [rn/view {:style (style/root-container (:top insets))}
+           [header active-tab read-qr-once? title]
+           (case @active-tab
+             1 [scan-qr-code-tab qr-view-finder request-camera-permission
```

Comment:
I think a bit better than `1` or `2` is to give meaningful names to the tabs, like `:tab/scan-qr-code` and `:tab/enter-sync-code`.

Reply:
Yep that's a good idea 👍